### PR TITLE
[Tooling] Automate Release Publishing

### PR DESCRIPTION
**Pull request by iangmaia**
This PR automates the release publishing, which means to:

- Publish the existing draft release on GitHub (which will also have GitHub create the associated git tag, pointing to the tip of the branch)
- If the release branch for the next version already exists, backmerge the current release into it
- Delete the current release branch

This PR will be followed by updates to the Releases V2 tool.

### CODE REVIEW

File: fastlane/lanes/release.rb [hunk 1]
```diff
@@ -435,6 +434,47 @@
...
+
+    UI.important <<~PROMPT
+      Publish the #{version_number} release. This will:
+      - Publish the existing draft `#{version_number}` release on GitHub
+      - Which will also have GitHub create the associated git tag, pointing to the tip of the branch
+      - If the release branch for the next version `#{next_release_branch}` already exists, backmerge `#{current_branch}` into it
+      - If needed, backmerge `#{current_branch}` back into `#{DEFAULT_BRANCH}`
+      - Delete the `#{current_branch}` branch
+    PROMPT
+    UI.user_error!("Terminating as requested. Don't forget to run the remainder of this automation manually.") unless skip_confirm || UI.confirm('Do you want to continue?')
```
Comment (dangermattic) [inline fastlane/lanes/release.rb:461]: <table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/WordPress-iOS/blob/d605d05475e9147e6b5a46744368c7ae28010d9a/fastlane/lanes/release.rb#L461"></span>Style/IfUnlessModifier: Modifier form of <code>unless</code> makes the line too long.</td>
    </tr>
  </tbody>
</table>

File: Gemfile.lock [hunk 2]
```diff
@@ -204,7 +204,7 @@ GEM
     fastlane-plugin-appcenter (2.1.2)
     fastlane-plugin-sentry (1.24.0)
       os (~> 1.1, >= 1.1.4)
-    fastlane-plugin-wpmreleasetoolkit (11.1.0)
+    fastlane-plugin-wpmreleasetoolkit (12.0.0)
```
Comment (spencertransier) [inline Gemfile.lock]: This could be updated to `12.1.0` while you're in here 🙂

**Review by spencertransier — APPROVED** [review]
:shipit: Just had one comment about updating the Release Toolkit from `12.0` to `12.1`

File: Gemfile.lock (hunk 2 above)
Comment (iangmaia) [reply Gemfile.lock]: Updated on 355c3d544a41fd1ca30221e6dc941b3fd81876eb

**Pull request merged**